src/core/solvers/box-ddp.cpp

```diff
@@ -89,16 +82,23 @@ void SolverBoxDDP::forwardPass(const double& steplength) {
   cost_try_ = 0.;
   xnext_ = problem_->get_x0();
   const std::size_t& T = problem_->get_T();
+  const std::vector<boost::shared_ptr<ActionModelAbstract> >& models = problem_->get_runningModels();
+  const std::vector<boost::shared_ptr<ActionDataAbstract> >& datas = problem_->get_runningDatas();
   for (std::size_t t = 0; t < T; ++t) {
-    const boost::shared_ptr<ActionModelAbstract>& m = problem_->get_runningModels()[t];
-    const boost::shared_ptr<ActionDataAbstract>& d = problem_->get_runningDatas()[t];
+    const boost::shared_ptr<ActionModelAbstract>& m = models[t];
+    const boost::shared_ptr<ActionDataAbstract>& d = datas[t];
+
     xs_try_[t] = xnext_;
     m->get_state()->diff(xs_[t], xs_try_[t], dx_[t]);
-    us_try_[t].noalias() = us_[t] - k_[t] * steplength - K_[t] * dx_[t];
-    if (m->get_has_control_limits()) {  // clamp control
-      us_try_[t] = us_try_[t].cwiseMax(m->get_u_lb()).cwiseMin(m->get_u_ub());
+    if (m->get_nu() != 0) {
+      us_try_[t].noalias() = us_[t] - k_[t] * steplength - K_[t] * dx_[t];
+      if (m->get_has_control_limits()) {  // clamp control
+        us_try_[t] = us_try_[t].cwiseMax(m->get_u_lb()).cwiseMin(m->get_u_ub());
+      }
+      m->calc(d, xs_try_[t], us_try_[t]);
```

Comment:
again, the solver should not care about what the dimension of actuation is. The API should remain the same here for both cases.